keps/sig-storage/1412-immutable-secrets-and-configmaps/README.md

```diff
@@ -197,6 +181,135 @@ immutable Secrets and/or ConfigMaps. That said, this is purely a performance
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: ImmutableEphemeralVolumes
+    - Components depending on the feature gate: kube-apiserver, kubelet
+
+* **Does enabling the feature change any default behavior?**
+  No, users need to explicitly opt-in for secrets/configmaps immutability.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Yes. kube-apiserver rollback allows for modifying secrets/configmaps.
+  kubelet rollback causes that all secrets/configmaps are again being watched.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  If the `immutable` field of Secret/ConfigMap wasn't cleared up for some
+  objects, this would result in making the immutable again.
+
+* **Are there any tests for feature enablement/disablement?**
+  Only basic conversions tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  If some critical Secrets/Configmaps were marked as immutable it may not be
+  able to update them if needed. No impact for already running workloads.
+
+* **What specific metrics should inform a rollback?**
+  No known rollback criteria.
+
+* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
+  Manual tests run to confirm that (on top of existing e2e tests):
+  - setting immutable field is impossible when the feature is switched of
+  - upgrade doesn't change behavior of existing objects
+  - immutable fields is dropped only on object touch after rollback
+  - on second upgrade, Secrets/Configmaps previously marked as immutable,
+    automatically are stopped being watched after Kubelet upgrade
+
+* **Is the rollout accompanied by any deprecations and/or removals of features,
+  APIs, fields of API types, flags, etc.?**
+  No.
+
+### Monitoring requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Listing secrets/configmaps in the cluster and checking if any has immutable 
+  field set.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to
+  determine the health of the service?**
+  None. It's not a feature, rather defence and optimization.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  None. It's not a feature, rather defence and optimization.
+
+* **Are there any missing metrics that would be useful to have to improve
+  observability if this feature?**
+  Number of currently in-use immutable secrets/configmaps in kubelet to be added.
```

Comment:
Would we see some indirect impact on existing metrics such as:

- kubelet cpu/memory usage
- apiserver requests
- kubelet volume mount operations

Reply:
for kubelet cpu/memory it should be negligible (we still store the current version in memory, we just don't watch for updates)

apiserver requests is a good one - added to the SLI section.

kubelet volume mount operations - actually I don't think it impacts it visible - the initial mounting still waits until the secret is fetched.  Do I miss anything?

Reply:
For the volume operations, would we still try to remount the secret/configmap every minute with this feature? We have a metric that measures the "MountVolume" call, and it gets called if we return true for [IsRemountRequired](https://github.com/kubernetes/kubernetes/blob/f52cbea102bb6ba9fe951cbf81e7dd4b48f95d9e/pkg/kubelet/volumemanager/reconciler/reconciler.go#L247)

Reply:
I didn't update this one (so yes, we are still trying to remount periodically).